### DISCUSSION TITLE
Missing documentation for razor tag helpers

### DISCUSSION THREAD

**Discussion by scleaver** [OrchardCMS/OrchardCore#16209] [Q&A]
I have been looking high and low for documentation on the OC razor tag helpers like shape, a tag etc.

There are a few scattered through the documentation on things like resources and media topics but not much else. Nothing as definitive as the liquid tag list.

I think it would be a great idea to put a dedicated page on the topic. I only know a few but recall seeing more on my travels so it would take some help from sone more knowledgeable people.

Thoughts?

etc – are you aware of any?
 
A list like the liquid ones would be really great I would think.
 
Seth

**Comment by Piedone** [comment]
Yeah, there are a lot more tag helpers than what we have documentation for, e.g.:

![image](https://github.com/OrchardCMS/OrchardCore/assets/1976647/bb0892c1-76cd-4355-a432-2ea90a97e0c2)

BTW the shape tag helper, specifically, is [kind of documented](https://docs.orchardcore.net/en/latest/reference/modules/DynamicCache/#shape-tag-helper-attributes). Documenting these looks like a good first task for a new contributor.

I don't think though that we'd need a separate list for all of them at once, since tag helpers are provided by specific modules, for specific scenarios, and they don't make sense for everyone in every app (or might not even be available, if you don't have the given module enabled).

So yeah, I'd document these under the specific features. Maybe we could even have a long list of them, but only linking to the feature docs pages. Could you please create an issue about this so we can track the task?

**Reply by scleaver** [reply]
I agree... I'll see what I can do.

That cache reference to shape helper assumes you know about the shape helper :D
